bindings/wasm/src/did/wasm_document.rs

```diff
@@ -87,10 +101,118 @@ impl WasmDocument {
+        None
+      } else {
+        Some(OneOrSet::try_from(OrderedSet::from_iter(controllers)).wasm_result()?)
+      }
+    } else {
+      None
+    };
+    *self.0.controller_mut() = controller_set;
+    Ok(())
+  }
+
+  /// Returns a list of document controllers.
+  ///
+  /// Note: Duplicates will be ignored.
+  /// Use `null` to remove all controllers.
+  #[wasm_bindgen]
+  pub fn controller(&self) -> ArrayDID {
+    match self.0.controller() {
+      Some(controllers) => controllers
+        .iter()
+        .cloned()
+        .map(WasmDID::from)
+        .map(JsValue::from)
+        .collect::<js_sys::Array>()
+        .unchecked_into::<ArrayDID>(),
+      None => js_sys::Array::new().unchecked_into::<ArrayDID>(),
+    }
+  }
+
+  /// Sets the `alsoKnownAs` property in the DID document.
+  #[wasm_bindgen(js_name = setAlsoKnownAs)]
+  pub fn set_also_known_as(&mut self, urls: &UOneOrManyUrl) -> Result<()> {
+    let urls: Option<OneOrMany<String>> = urls.into_serde().wasm_result()?;
+    let mut urls_set: OrderedSet<Url> = OrderedSet::new();
+    if let Some(urls) = urls {
+      for url in urls.into_vec() {
+        urls_set.append(Url::parse(url).wasm_result()?);
+      }
+    }
+    *self.0.also_known_as_mut() = urls_set;
+    Ok(())
+  }
+
+  /// Returns a set of the document's `alsoKnownAs`.
+  #[wasm_bindgen(js_name = alsoKnownAs)]
+  pub fn also_known_as(&self) -> ArrayString {
+    self
+      .0
+      .also_known_as()
+      .iter()
+      .map(|url| url.to_string())
+      .map(JsValue::from)
+      .collect::<js_sys::Array>()
+      .unchecked_into::<ArrayString>()
+  }
+
+  /// Adds a custom property to the DID Document.
+  /// If the value is set to `null`, the custom property will be removed.
+  ///
+  /// ### WARNING
+  /// This method can overwrite existing properties like `id` and result in an invalid document.
+  #[wasm_bindgen(js_name = setPropertyUnchecked)]
+  pub fn set_property_unchecked(&mut self, key: String, value: &JsValue) -> Result<()> {
+    let value: Option<serde_json::Value> = value.into_serde().wasm_result()?;
+    match value {
+      Some(value) => {
+        self.0.properties_mut().insert(key, value);
+      }
+      None => {
+        self.0.properties_mut().remove(&key);
+      }
+    }
+    Ok(())
+  }
+
+  /// Returns the custom DID Document properties.
+  #[wasm_bindgen]
+  pub fn properties(&mut self) -> Result<MapStringAny> {
+    let properties_map = js_sys::Map::new();
+    for (key, value) in self.0.properties().iter() {
+      properties_map.set(&JsValue::from(key), &JsValue::from_serde(&value).wasm_result()?);
+    }
+    Ok(properties_map.unchecked_into::<MapStringAny>())
+  }
+
   // ===========================================================================
   // Services
   // ===========================================================================
 
+  /// Return a set of all `Service`s in the document.
+  #[wasm_bindgen]
+  pub fn service(&self) -> ArrayService {
```

Comment:
Is there a way for the doc-comment to link to `Service` but replace the content with `Services`? Like we do in Rust: `[Services](Service)`.

```suggestion
  /// Return a set of all {@link Service}s in the document.
  #[wasm_bindgen]
  pub fn service(&self) -> ArrayService {
```

Reply:
Yes, `{@link Service Services}` achieves exactly that.